gwctl/pkg/printer/gatewayclasses.go

```diff
@@ -17,37 +17,87 @@ limitations under the License.
 package printer
 
 import (
+	"encoding/json"
 	"fmt"
 	"io"
+	"k8s.io/apimachinery/pkg/util/duration"
 	"os"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/yaml"
 	"sort"
 	"strings"
 	"text/tabwriter"
 
-	"sigs.k8s.io/yaml"
-
 	"sigs.k8s.io/gateway-api/gwctl/pkg/policymanager"
 	"sigs.k8s.io/gateway-api/gwctl/pkg/resourcediscovery"
 
-	"k8s.io/apimachinery/pkg/util/duration"
 	"k8s.io/utils/clock"
 )
 
+var _ Printer = (*GatewayClassesPrinter)(nil)
+
 type GatewayClassesPrinter struct {
 	Out   io.Writer
 	Clock clock.Clock
 }
 
-type gatewayClassDescribeView struct {
-	// GatewayClass name
-	Name           string `json:",omitempty"`
-	ControllerName string `json:",omitempty"`
-	// GatewayClass description
-	Description              *string                `json:",omitempty"`
-	DirectlyAttachedPolicies []policymanager.ObjRef `json:",omitempty"`
+func (gcp *GatewayClassesPrinter) PrintJson(resourceModel *resourcediscovery.ResourceModel) {
+	gatewayClassNodes := make([]*resourcediscovery.GatewayClassNode, 0, len(resourceModel.GatewayClasses))
+	for _, gatewayClassNode := range resourceModel.GatewayClasses {
+		gatewayClassNodes = append(gatewayClassNodes, gatewayClassNode)
+	}
+
+	sort.Slice(gatewayClassNodes, func(i, j int) bool {
+		if gatewayClassNodes[i].GatewayClass.GetName() != gatewayClassNodes[j].GatewayClass.GetName() {
+			return gatewayClassNodes[i].GatewayClass.GetName() < gatewayClassNodes[j].GatewayClass.GetName()
+		}
+		return string(gatewayClassNodes[i].GatewayClass.Spec.ControllerName) < string(gatewayClassNodes[j].GatewayClass.Spec.ControllerName)
```

Comment:
Do we really need such a fallback on the `controllerName`? As far as I understand, the only case to have the `GatewayClass`es with the same name is that `i==j`; in that case we can just skip while for all other cases the `GatewayClass.Name`s will always be different, as the `GatewayClass` is a cluster-wide resource.

Reply:
You're right, let's get rid of that!